airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py

```diff
@@ -155,3 +168,343 @@ def cache_filename(self) -> str:
     def use_cache(self) -> bool:
         # FIXME: this should be declarative
         return False
+
+    disable_retries: bool = False
+    _DEFAULT_MAX_RETRY = 5
+    _DEFAULT_RETRY_FACTOR = 5
+
+    @property
+    def max_retries(self) -> Union[int, None]:
+        if self.disable_retries:
+            return 0
+        if hasattr(self.error_handler, "max_retries"):
```

Comment:
I would expect `max_retries` to always be set since it is part of the ErrorHandler interface

Reply:
I can return None though - I will adjust